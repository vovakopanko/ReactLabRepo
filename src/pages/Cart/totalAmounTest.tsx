type imagesPlatforms = { id: number; src: string; alt: string };

type cartList = {
  name: string;
  platforms: imagesPlatforms[];
  orderDate: string;
  amount: number;
  price: number;
  checked: boolean;
};

export const totalAmount = (selectedGames: cartList[]) => {
  return Number(
    selectedGames
      .reduce((prev, cardItem) => {
        if (cardItem.checked) {
          return prev + cardItem.price * cardItem.amount;
        }
        return prev;
      }, 0)
      .toFixed(2)
  );
};
