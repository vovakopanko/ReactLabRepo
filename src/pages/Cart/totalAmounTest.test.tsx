import { totalAmount } from "./totalAmounTest";

const selectedGame = [
  {
    amount: 1,
    checked: true,
    name: "Terraria",
    orderDate: "9/5/2022",
    platforms: [
      {
        alt: "Xbox",
        id: 0,
        src: "https://upload.wikimedia.org/wikipedia/commons/thumb/f/f9/Xbox_one_logo.svg/1024px-Xbox_one_logo.svg.png",
      },
    ],
    price: 4.87,
  },
];

const selectedGames = [
  {
    amount: 1,
    checked: true,
    name: "Terraria",
    orderDate: "9/5/2022",
    platforms: [
      {
        alt: "Xbox",
        id: 0,
        src: "https://upload.wikimedia.org/wikipedia/commons/thumb/f/f9/Xbox_one_logo.svg/1024px-Xbox_one_logo.svg.png",
      },
    ],
    price: 4.87,
  },
  {
    amount: 2,
    checked: true,
    name: "SIMS 4",
    orderDate: "9/5/2022",
    platforms: [
      {
        alt: "Xbox",
        id: 0,
        src: "https://upload.wikimedia.org/wikipedia/commons/thumb/f/f9/Xbox_one_logo.svg/1024px-Xbox_one_logo.svg.png",
      },
    ],
    price: 5.01,
  },
];

describe("Test how work amount counter", () => {
  test("Test counter work correct", () => {
    expect(totalAmount(selectedGames)).toBe(14.89);
  });
  test("Test how work amount counter", () => {
    expect(totalAmount(selectedGame)).toBe(4.87);
  });
  test("The test counter is not working properly", () => {
    expect(totalAmount(selectedGames)).not.toBe(14.88);
  });
  test("The test counter is not working properly", () => {
    expect(totalAmount(selectedGame)).not.toBe(4.89);
    expect(totalAmount(selectedGame)).toMatchSnapshot();
  });
});
