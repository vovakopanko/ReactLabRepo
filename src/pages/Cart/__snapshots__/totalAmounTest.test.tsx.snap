// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test how work amount counter The test counter is not working properly 1`] = `4.87`;
